ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java

```diff
@@ -0,0 +1,249 @@
+
+    private static final Vector3f WORLD_LOCATION = new Vector3f(300, -20, 0);
+
+    private static final Vector3f PLATFORM_LOCATION = new Vector3f(20, -18, -1);
+    private static final Vector3f COMMANDER_LOCATION = new Vector3f(23, -14, -1f);
+
+    private static final float COMMANDER_ROTATION = -1.5f;
+
+    private Platform platform;
+    private Commander commander;
+
+    private DirectionalLight sun;
+    private DirectionalLight sun2;
+
+    private float steering;
+
+    private boolean flyCamActive;
+
+    private LevelGenerator levelGenerator;
+
+    /**
+     * First method that is called after the state has been created.
+     * Handles all initialization of parameters needed for the Environment.
+     */
+    @Override
+    public void initialize(AppStateManager stateManager, Application app) {
+        super.initialize(stateManager, app);
+
+        viewPort = app.getViewPort();
+        flyObs = new Camera();
+        steering = 0.f;
+        flyCamActive = false;
+
+        viewPort.setBackgroundColor(BACKGROUNDCOLOR);
+
+        //creates the lights
+        initLights();
+
+        //creates shadowmap and attach it to the sun
+        initShadows();
+
+        //initializes all spatials and set the positions
+        initSpatials();
+
+        //Init the camera
+        initCameras();
+    }
+    
+    @Override
+    public void update(float tpf) {
+    	super.update(tpf);
+    	
+    	updateTestWorld();
+    }
+
+    /**
+     * Initializes all lights of the scene.
+     */
+    private void initLights() {
+        sun = new DirectionalLight();
+        sun2 = new DirectionalLight();
+
+        sun.setColor(ColorRGBA.White);
+        sun.setDirection(new Vector3f(SUNVECTOR).normalizeLocal());
+
+        sun2.setColor(ColorRGBA.White);
+        sun2.setDirection(new Vector3f(SUNVECTOR_2).normalizeLocal());
+
+        //adds the lights to the root pane
+        getRootNode().addLight(sun);
+        getRootNode().addLight(sun2);
+    }
+
+    /**
+     * Initializes all shadows of the scene.
+     */
+    private void initShadows() {
+        DirectionalLightShadowRenderer dlsr =
+                new DirectionalLightShadowRenderer(getAssetManager(), SHADOWMAP_SIZE, SHADOW_SPLITS);
+        dlsr.setLight(sun);
+        viewPort.addProcessor(dlsr);
+
+        //adds shadow filter and attaches it to the viewport
+        DirectionalLightShadowFilter dlsf =
+                new DirectionalLightShadowFilter(getAssetManager(), SHADOWMAP_SIZE, SHADOW_SPLITS);
+        dlsf.setLight(sun);
+        dlsf.setEnabled(true);
+        FilterPostProcessor fpp = new FilterPostProcessor(getAssetManager());
+        fpp.addFilter(dlsf);
+        viewPort.addProcessor(fpp);
+    }
+
+    /**
+     * Initializes all objects and translations/rotations of the scene.
+     */
+    private void initSpatials() {
+
+        levelGenerator = new LevelGenerator(WORLD_LOCATION);
+        platform = new Platform(PLATFORM_LOCATION);
```

Comment:
I didn't write the description :p

But we certainly do NOT want this in the superclass Environment. Because you'd just be hardcoding the platform and other spatials in it, while the Environment class really is supposed to be a general Environment, that can then be extended to create a specific hardcoded environment like MainEnvironment.